src/db/models/ApplicationModel.ts

```diff
@@ -75,4 +79,36 @@ export class ApplicationModel extends Model<ApplicationModel> {
     onDelete: "RESTRICT",
   })
   resume: ResumeModel;
+
+  @AfterUpdate
+  static async sendEmailOnEventChange(instance: ApplicationModel) {
+    // Check if the eventId has been modified
+    if (instance.previous("eventId") !== instance.eventId) {
+      const mailerService = new MailerService();
+
+      // Retrieve the associated student details
+      const student = await StudentModel.findByPk(instance.studentId);
```

Comment:
Use include for fetching associated models using eager loading this will be faster and will have lesser queries to the database.

Example:
```js
const users = await User.findAll({
  include: { model: Tool, as: 'Instruments' },
});
```